api/filter/accesslog.proto

```diff
@@ -53,120 +80,126 @@ message AccessLogEntry {
     // Request was ratelimited locally by rate limit filter.
     RATE_LIMITED = 11;
   }
-  // Status flags about the response.
-  repeated ResponseFlag response_flags = 4;
+  repeated ResponseFlag response_flags = 10;
 
-  // The upstream host URL (Envoy connects to).
-  //
-  // For example, tcp://ip:port for TCP connections.
+  // All metadata encountered during request processing, including endpoint
+  // selection.
   //
-  // IPv6 addresses should be stored in canonical (compressed) format using
-  // [address]:port notation.
-  string upstream_host = 5;
+  // This can be used to associate IDs attached to the various configurations
+  // used to process this request with the access log entry.  For example, a
+  // route created from a higher level forwarding rule with some ID can place
+  // that ID in this field and cross reference later. It can also be used to
+  // determine if a canary endpoint was used or not.
+  Metadata metadata = 11;
+}
 
-  // The Upstream Cluster that the upstream host belongs to.
-  string upstream_cluster = 6;
+message TLSProperties {
+  // TLS Version
+  enum TLSVersion {
```

Comment:
I mean it does look like there's a fair amount of overlap, but I'm always a little bit cautious about sharing structures across disjoint concerns. If configuration requires a bunch of extra new fields to be part of a shared TLSProperties message, then all of those end up in the log format as well. And vice versa.

But really just a little bit cautious. Maybe there's a good case to be made for it here. Maybe both contexts really do want the same properties. Or maybe the the structure is unlikely to change significantly. Or maybe it's fine regardless. ¯\\\_(ツ)\_/¯

Reply:
I think it would work well as a shared message here, as it's essentially reflecting what we're using in Listener configuration. That said, let's not block this PR merge and we can clean it up a bit later, before we promote the access log work from draft to frozen.